provisioning/device/samples/ProvisioningDeviceClientTpm/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Provisioning.Client;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport;
+using Microsoft.Azure.Devices.Provisioning.Security.Samples;
+using System;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ProvisioningDeviceClientTpm
+{
+    public static class Program
+    {
+        private static string s_idScope;
+        private const string RegistrationId = "testtpmregistration1";
+
+        public static async Task RunSample()
+        {
+            // Replace the following type with SecurityClientTpm() to use a real TPM2.0 device.
+            Console.WriteLine("Starting TPM simulator.");
+            SecurityClientTpmSimulator.StartSimulatorProcess();
+
+            using (var security = new SecurityClientTpmSimulator(RegistrationId))
+            using (var transport = new ProvisioningTransportHandlerHttp())
+            {
+
+                // Note that the TPM simulator will create a NVChip file containing the simulated TPM state.
+                Console.WriteLine("Extracting endorsement key.");
+                string base64EK = Convert.ToBase64String(security.GetEndorsementKey());
+
+                Console.WriteLine(
+                    "In your Azure Device Provisioning Service please go to 'Manage enrollments' and select " +
+                    "'Individual Enrollments'. Select 'Add' then fill in the following:");
```

Comment:
Is this required or can you do the whole flow besides initial creation of the resource via code?

Reply:
Before this sample runs, it's required that the device is known by the service. That can be achieved by either the actions described above or, programmatic via ProvisioningServiceClient. These actions are expected to be performed in the hardware manufacturing factory.